sysbuild/Kconfig.matter

```diff
@@ -37,7 +37,22 @@ config MATTER_OTA_IMAGE_FILE_NAME
 	help
 	  Provides the file name of the generated Matter OTA image in the build directory.
 
+if SOC_SERIES_NRF54HX
+
+config SUIT_ENVELOPE
+	default y
+
+config SUIT_BUILD_FLASH_COMPANION
+	default y
+
+config SUIT_MULTI_IMAGE_PACKAGE_BUILD
+	default y if MATTER_OTA
+
 config SUIT_BASE_MANIFEST_VARIANT
 	default "matter/v1"
 
-endif
+config SUIT_ENVELOPE_NORDIC_TOP_EXTRACT_PAYLOADS_TO_CACHE
+	default y if SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY != ""
```

Comment:
should not be abusing Kconfig files as thought they are defconfig files. If these are needed then the symbols should depend on them, or select them if the known issues are adhered to properly

Reply:
Hmm, I would like to set the `SUIT_ENVELOPE_NORDIC_TOP_EXTRACT_PAYLOADS_TO_CACHE` if the user provides a path to the top directory using `SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY`. Currently in `Kconfig.suit` `SUIT_ENVELOPE_NORDIC_TOP_EXTRACT_PAYLOADS_TO_CACHE` depends on `SUIT_ENVELOPE_NORDIC_TOP_IN_ROOT` and `SUIT_ENVELOPE_NORDIC_TOP_IN_ROOT` is default `y` if SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY != "". I don't understand the abuse here, which kconfig of all mentioned should select or have dependency?

Reply:
Thanks for the explanation. I've modified them. Indeed if we use MATTER sysbuild config at least one DFU method must be present (according to spec). So we must select SUIT mechanisms for the nRF54H20 device anyway and forbid users to change them. There is only one exception - `SUIT_BASE_MANIFEST_VARIANT` which may be different in a user application. Could you please take a look at whether now they're set properly?

Reply:
@nordicjm Is it OK now after the recent change?